versionparse/lib.go

```diff
@@ -20,47 +20,158 @@ import (
 	"io/ioutil"
 	"os"
 	"regexp"
+	"strconv"
+	"strings"
 
-	version "github.com/hashicorp/go-version"
 	log "github.com/sirupsen/logrus"
 )
 
 var (
-	kernelVersionRegexp = regexp.MustCompile(`Linux version (\d+\.\d+\.\d+)`)
+	kernelVersionRegexp     = regexp.MustCompile(`Linux version (\d+\.\d+\.\d+)`)
+	kernelVersionRHELRegexp = regexp.MustCompile(`Linux version (\d+\.\d+\.\d-\d+)`)
```

Comment:
Are we sure that RH always append a `-%d+` on their versions?  What if they ship a vanilla kernel as the first one?  Think I'd make the regex generic so it can match either type:
```
regexp.MustCompile(`Linux version (\d+\.\d+\.\d+(?:-\d+)?)`)
```